diff --git a/lib/python/qtvcp/widgets/jog_increments.py b/lib/python/qtvcp/widgets/jog_increments.py
index 1272c5662..2792999ab 100644
--- a/lib/python/qtvcp/widgets/jog_increments.py
+++ b/lib/python/qtvcp/widgets/jog_increments.py
@@ -56,7 +56,8 @@ class JogIncrements(QtWidgets.QComboBox, _HalWidgetBase):
         self.selectionchange(0)
 
     def _switch_units(self, w, data):
-        self.selectionchange(-1)
+        if self.currentIndex() > 0:
+            self.selectionchange(-1)
 
     # search the combo box for the value STATUS sent us
     # If there is a match, change the combobox to it
@@ -66,7 +67,11 @@ class JogIncrements(QtWidgets.QComboBox, _HalWidgetBase):
     def _checkincrements(self,value, text):
         for count in range(self.count()):
             label = self.itemText(count)
-            if 'cont' in label.lower():
+            try:
+                number = float(label.rstrip(" inchmuil").lower())
+            except:
+                number = None
+            if not number:
                 machn_incr = 0
             else:
                 if self.linear:
@@ -86,7 +91,7 @@ class JogIncrements(QtWidgets.QComboBox, _HalWidgetBase):
         text = str(self.currentText())
         if i == -1 and text == '': return
         try:
-            if 'cont' in text.lower():
+            if i == 0:
                 inc = 0
             else:
                 if self.linear:
